{{cookiecutter.project_slug}}/Makefile

```diff
@@ -331,6 +335,60 @@ endif
 kill-train:  ### Terminate the training job (set up env var 'RUN' to specify the training job)
 	$(NEURO) kill $(TRAIN_JOB)-$(RUN) || :
 
+# Number of hyper-parameter search jobs
+N_HYPERPARAM_JOBS?=3
+
+# Sweep ID (if empty, will be created during 'make hypertrain')
+SWEEP?=
+SWEEP_REAL:=$$([ "${SWEEP}" ] && echo "${SWEEP}" || tail -1 ${WANDB_SWEEPS_FILE})
+
+.PHONY: hypertrain
+hypertrain: _check_setup wandb-check-auth   ### Run jobs in parallel for hyperparameters search using W&B
+	@echo "Initializing local wandb using config file './${WANDB_SECRET_PATH_LOCAL}'"
+	@wandb login $$(cat "./${WANDB_SECRET_PATH_LOCAL}")
+	@[ "${SWEEP}" ] || { \
+		echo "Creating W&B Sweep..." && \
+		echo "Using variable: WANDB_SWEEP_CONFIG_FILE='${WANDB_SWEEP_CONFIG_FILE}'" && \
+		[ -f "${WANDB_SWEEP_CONFIG_PATH}" ] \
+			&& echo "Using W&B sweep file: ./${WANDB_SWEEP_CONFIG_PATH}" \
+			|| { echo "ERROR: W&B sweep config file not found: '$${PWD}/${WANDB_SWEEP_CONFIG_PATH}'" >&2; false ; \
+		}; \
+		wandb sweep ${WANDB_SWEEP_CONFIG_PATH} 2>&1 | tee /dev/stderr | grep 'Created sweep with ID: ' | awk 'NF{ print $$NF }' >> ${WANDB_SWEEPS_FILE} ; \
+    }
+	@[ "${SWEEP_REAL}" ] || { echo "Error while getting sweep ID, please set up env var 'SWEEP' manually." >&2; false; }
+	$(NEURO) mkdir -p $(RESULTS_DIR_STORAGE)/sweep-${SWEEP_REAL}
+	@echo "Uploading wandb config file './wandb/settings' to Neuro storage..."
+	$(NEURO) mkdir -p $(PROJECT_PATH_STORAGE)/wandb
+	$(NEURO) cp ./wandb/settings $(PROJECT_PATH_STORAGE)/wandb/
+	@echo "Running ${N_HYPERPARAM_JOBS} jobs of sweep ${SWEEP_REAL}..."
+	for i in $$(seq 1 ${N_HYPERPARAM_JOBS}) ; do \
+		echo -e "\nStarting job $${i}..." ; \
+		make train \
+			TRAIN_CMD="wandb agent ${SWEEP_REAL}" \
+			RUN=${SWEEP_REAL}-$${i} \
+			RESULTS_DIR_STORAGE=$(RESULTS_DIR_STORAGE)/sweep-${SWEEP_REAL} \
+			_SWEEP_DESCRIPTION_POSTFIX=":${SWEEP_REAL}" \
+			_EXTRA_VOLUME_OPTION="--volume $(PROJECT_PATH_STORAGE)/wandb:$(PROJECT_PATH_ENV)/wandb:rw" \
```

Comment:
If we don't put the file `./wandb/settings` created after `wandb sweep ...`, wandb won't use the correct project (and possibly other settings stored in this file), so the command `wandb agent <swipe>` won't work from within the job.